Bottom: 4e4e427c500e8706ebc472c0918400c714625c59
Top:    65aaf0c68aa0ad4174edd8336e8902c11b022d24
Author: Andrew Cooper <andrew.cooper@novatechweb.com>
Date:   2018-05-25 10:34:57 -0500

fixup! tmp-rework-ip-address-network


---
diff --git a/ansible-playbook/group_vars/docker-hosts/networking.yml b/ansible-playbook/group_vars/docker-hosts/networking.yml
new file mode 100644
index 00000000..77d9ce6a
--- /dev/null
+++ b/ansible-playbook/group_vars/docker-hosts/networking.yml
@@ -0,0 +1,66 @@
+---
+
+svn_ip_addr: "{{ daedalus_subnet | ipaddr('101') | ipaddr('address') }}"
+svn_hostname: svn.novatech-llc.com
+svn_port_args:
+- "{{ svn_ip_addr }}:80:80"
+- "{{ svn_ip_addr }}:443:443"
+
+gitlab_ip_addr: "{{ daedalus_subnet | ipaddr('102') | ipaddr('address') }}"
+gitlab_hostname: git.novatech-llc.com
+gitlab_port_args:
+- "{{ gitlab_ip_addr }}:80:80"
+- "{{ gitlab_ip_addr }}:443:443"
+- "{{ gitlab_ip_addr }}:22:22"
+
+wiki_ip_addr: "{{ daedalus_subnet | ipaddr('103') | ipaddr('address') }}"
+wiki_hostname: wiki.novatech-llc.com
+wiki_port_args:
+- "{{ wiki_ip_addr }}:80:80"
+- "{{ wiki_ip_addr }}:443:443"
+
+mantisbt_ip_addr: "{{ daedalus_subnet | ipaddr('104') | ipaddr('address') }}"
+mantisbt_hostname: mantis.novatech-llc.com
+mantisbt_port_args:
+- "{{ mantisbt_ip_addr }}:80:80"
+- "{{ mantisbt_ip_addr }}:443:443"
+
+build_ip_addr: "{{ daedalus_subnet | ipaddr('105') | ipaddr('address') }}"
+build_hostname: buildsystem.novatech-llc.com
+build_port_args:
+- "{{ build_ip_addr }}:80:80"
+- "{{ build_ip_addr }}:443:443"
+
+testssh_ip_addr: "{{ daedalus_subnet | ipaddr('106') | ipaddr('address') }}"
+testssh_hostname: testssh.novatech-llc.com
+testssh_port_args:
+- "{{ testssh_ip_addr }}:2200:22"
+
+exim4_ip_addr: "{{ daedalus_subnet | ipaddr('107') | ipaddr('address') }}"
+exim4_hostname: mail.novatech-llc.com
+exim4_port_args:
+- "{{ exim4_ip_addr }}:25:25"
+- "{{ exim4_ip_addr }}:465:465"
+- "{{ exim4_ip_addr }}:587:587"
+
+openldap_ip_addr: "{{ daedalus_subnet | ipaddr('108') | ipaddr('address') }}"
+openldap_hostname: ldap.novatech-llc.com
+openldap_port_args:
+- "{{ openldap_ip_addr }}:389:389"
+- "{{ openldap_ip_addr }}:636:636"
+
+phpldapadmin_ip_addr: "{{ daedalus_subnet | ipaddr('108') | ipaddr('address') }}"
+phpldapadmin_hostname: ldap.novatech-llc.com
+phpldapadmin_port_args:
+- "{{ phpldapadmin_ip_addr }}:80:80"
+- "{{ phpldapadmin_ip_addr }}:443:443"
+
+phpmyadmin_ip_addr: "{{ daedalus_subnet | ipaddr('109') | ipaddr('address') }}"
+phpmyadmin_hostname: phpmyadmin.novatech-llc.com
+
+buildbot_ip_addr: "{{ daedalus_subnet | ipaddr('110') | ipaddr('address') }}"
+buildbot_hostname: buildbot.novatech-llc.com
+buildbot_port_args:
+- "{{ buildbot_ip_addr }}:80:8080"
+- "{{ buildbot_ip_addr }}:443:8443"
+- "{{ buildbot_ip_addr }}:9989:9989"
diff --git a/ansible-playbook/host_vars/daedalus.novatech-llc.com/networking.yml b/ansible-playbook/host_vars/daedalus.novatech-llc.com/networking.yml
index 3434cb2e..f567d255 100644
--- a/ansible-playbook/host_vars/daedalus.novatech-llc.com/networking.yml
+++ b/ansible-playbook/host_vars/daedalus.novatech-llc.com/networking.yml
@@ -1,39 +1,4 @@
 ---
 
-svn_network:
-  ip_addr: 172.16.0.101
-  hostname: svn.novatech-llc.com
-gitlab_network:
-  ip_addr: 172.16.0.102
-  hostname: git.novatech-llc.com
-wiki_network:
-  ip_addr: 172.16.0.103
-  hostname: wiki.novatech-llc.com
-mantisbt_network:
-  ip_addr: 172.16.0.104
-  hostname: mantis.novatech-llc.com
-build_network:
-  ip_addr: 172.16.0.105
-  hostname: buildsystem.novatech-llc.com
-testssh_network:
-  ip_addr: 172.16.0.106
-  hostname: testssh.novatech-llc.com
-exim4_network:
-  ip_addr: 172.16.0.107
-  hostname: mail.novatech-llc.com
-openldap_network:
-  ip_addr: 172.16.0.108
-  hostname: ldap.novatech-llc.com
-phpldapadmin_network:
-  ip_addr: 172.16.0.108
-  hostname: ldap.novatech-llc.com
-phpmyadmin_network:
-  ip_addr: 172.16.0.109
-  hostname: phpmyadmin.novatech-llc.com
-
-buildbot_network:
-  ip_addr: 172.16.0.110
-  hostname: buildbot.novatech-llc.com
-  port_args:
-    - '{{ buildbot_network.ip_addr }}:80:80'
-    - '{{ buildbot_network.ip_addr }}:443:443'
+daedalus_subnet: 172.16.0.0/16
+daedalus_net_connection: "Bridge br0"
diff --git a/ansible-playbook/host_vars/testdaedalus.novatech-llc.com/networking.yml b/ansible-playbook/host_vars/testdaedalus.novatech-llc.com/networking.yml
index 9fd2bd1c..f55400be 100644
--- a/ansible-playbook/host_vars/testdaedalus.novatech-llc.com/networking.yml
+++ b/ansible-playbook/host_vars/testdaedalus.novatech-llc.com/networking.yml
@@ -1,76 +1,4 @@
 ---
 
-svn_network:
-  ip_addr: 172.16.0.201
-  hostname: svn.novatech-llc.com
-  port_args:
-    - '{{ svn_network.ip_addr }}:80:80'
-    - '{{ svn_network.ip_addr }}:443:443'
-
-gitlab_network:
-  ip_addr: 172.16.0.202
-  hostname: git.novatech-llc.com
-  port_args:
-    - '{{ gitlab_network.ip_addr }}:80:80'
-    - '{{ gitlab_network.ip_addr }}:443:443'
-    - '{{ gitlab_network.ip_addr }}:22:22'
-
-wiki_network:
-  ip_addr: 172.16.0.203
-  hostname: wiki.novatech-llc.com
-  port_args:
-    - '{{ wiki_network.ip_addr }}:80:80'
-    - '{{ wiki_network.ip_addr }}:443:443'
-
-mantisbt_network:
-  ip_addr: 172.16.0.204
-  hostname: mantis.novatech-llc.com
-  port_args:
-    - '{{ mantisbt_network.ip_addr }}:80:80'
-    - '{{ mantisbt_network.ip_addr }}:443:443'
-
-build_network:
-  ip_addr: 172.16.0.205
-  hostname: buildsystem.novatech-llc.com
-  port_args:
-    - '{{ build_network.ip_addr }}:80:80'
-    - '{{ build_network.ip_addr }}:443:443'
-
-testssh_network:
-  ip_addr: 172.16.0.206
-  hostname: testssh.novatech-llc.com
-  port_args:
-    - '{{ testssh_network.ip_addr }}:2200:22'
-
-exim4_network:
-  ip_addr: 172.16.0.207
-  hostname: mail.novatech-llc.com
-  port_args:
-    - '{{ exim4_network.ip_addr }}:25:25'
-    - '{{ exim4_network.ip_addr }}:465:465'
-    - '{{ exim4_network.ip_addr }}:587:587'
-
-openopenldap_network:
-  ip_addr: 172.16.0.208
-  hostname: ldap.novatech-llc.com
-  port_args:
-    - '{{ openopenldap_network.ip_addr }}:389:389'
-    - '{{ openopenldap_network.ip_addr }}:636:636'
-
-phpldapadmin_network:
-  ip_addr: 172.16.0.208
-  hostname: ldap.novatech-llc.com
-  port_args:
-    - '{{ phpldapadmin_network.ip_addr }}:80:80'
-    - '{{ phpldapadmin_network.ip_addr }}:443:443'
-
-phpmyadmin_network:
-  ip_addr: 172.16.0.209
-  hostname: phpmyadmin.novatech-llc.com
-
-buildbot_network:
-  ip_addr: 172.16.0.210
-  hostname: buildbot.novatech-llc.com
-  port_args:
-    - '{{ buildbot_network.ip_addr }}:80:80'
-    - '{{ buildbot_network.ip_addr }}:443:443'
+daedalus_subnet: 172.16.103.0/16
+daedalus_net_connection: "Bridge br0"
diff --git a/ansible-playbook/host_vars/vmdaedalus.novatech-llc.com/docker-container_addr_map-and-container_port_map.yml b/ansible-playbook/host_vars/vmdaedalus.novatech-llc.com/docker-container_addr_map-and-container_port_map.yml
deleted file mode 100644
index 53b96f84..00000000
--- a/ansible-playbook/host_vars/vmdaedalus.novatech-llc.com/docker-container_addr_map-and-container_port_map.yml
+++ /dev/null
@@ -1,33 +0,0 @@
----
-
-container_addr_map:
-  svn:
-    ip_addr: 192.168.0.101
-    hostname: svn.novatech-llc.com
-  git:
-    ip_addr: 192.168.0.102
-    hostname: git.novatech-llc.com
-  wiki:
-    ip_addr: 192.168.0.103
-    hostname: wiki.novatech-llc.com
-  mantisbt:
-    ip_addr: 192.168.0.104
-    hostname: mantis.novatech-llc.com
-  build:
-    ip_addr: 192.168.0.105
-    hostname: buildsystem.novatech-llc.com
-  testssh:
-    ip_addr: 192.168.0.106
-    hostname: testssh.novatech-llc.com
-  exim4:
-    ip_addr: 192.168.0.107
-    hostname: mail.novatech-llc.com
-  ldap:
-    ip_addr: 192.168.0.108
-    hostname: ldap.novatech-llc.com
-  phpldapadmin:
-    ip_addr: 192.168.0.108
-    hostname: ldap.novatech-llc.com
-  phpmyadmin:
-    ip_addr: 192.168.0.109
-    hostname: phpmyadmin.novatech-llc.com
diff --git a/ansible-playbook/host_vars/vmdaedalus.novatech-llc.com/networking.yml b/ansible-playbook/host_vars/vmdaedalus.novatech-llc.com/networking.yml
index 79bbb96d..bdc04890 100644
--- a/ansible-playbook/host_vars/vmdaedalus.novatech-llc.com/networking.yml
+++ b/ansible-playbook/host_vars/vmdaedalus.novatech-llc.com/networking.yml
@@ -1,65 +1,4 @@
 ---
 
-svn_ip_addr: 192.168.0.101/16
-svn_hostname: svn.novatech-llc.com
-svn_port_args:
-    - "{{ svn_ip_addr | ipaddr('address') }}:80:80"
-    - "{{ svn_ip_addr | ipaddr('address') }}:443:443"
-
-gitlab_ip_addr: 192.168.0.102/16
-gitlab_hostname: git.novatech-llc.com
-gitlab_port_args:
-    - "{{ gitlab_ip_addr | ipaddr('address') }}:80:80"
-    - "{{ gitlab_ip_addr | ipaddr('address') }}:443:443"
-    - "{{ gitlab_ip_addr | ipaddr('address') }}:22:22"
-
-wiki_ip_addr: 192.168.0.103/16
-wiki_hostname: wiki.novatech-llc.com
-wiki_port_args:
-    - "{{ wiki_ip_addr | ipaddr('address') }}:80:80"
-    - "{{ wiki_ip_addr | ipaddr('address') }}:443:443"
-
-mantisbt_ip_addr: 192.168.0.104/16
-mantisbt_hostname: mantis.novatech-llc.com
-mantisbt_port_args:
-    - "{{ mantisbt_ip_addr | ipaddr('address') }}:80:80"
-    - "{{ mantisbt_ip_addr | ipaddr('address') }}:443:443"
-
-build_ip_addr: 192.168.0.105/16
-build_hostname: buildsystem.novatech-llc.com
-build_port_args:
-    - "{{ build_ip_addr | ipaddr('address') }}:80:80"
-    - "{{ build_ip_addr | ipaddr('address') }}:443:443"
-
-testssh_ip_addr: 192.168.0.106/16
-testssh_hostname: testssh.novatech-llc.com
-testssh_port_args:
-    - "{{ testssh_ip_addr | ipaddr('address') }}:2200:22"
-
-exim4_ip_addr: 192.168.0.107/16
-exim4_hostname: mail.novatech-llc.com
-exim4_port_args:
-    - "{{ exim4_ip_addr | ipaddr('address') }}:25:25"
-    - "{{ exim4_ip_addr | ipaddr('address') }}:465:465"
-    - "{{ exim4_ip_addr | ipaddr('address') }}:587:587"
-
-openldap_ip_addr: 192.168.0.108/16
-openldap_hostname: ldap.novatech-llc.com
-openldap_port_args:
-    - "{{ openldap_ip_addr | ipaddr('address') }}:389:389"
-    - "{{ openldap_ip_addr | ipaddr('address') }}:636:636"
-
-phpldapadmin_ip_addr: 192.168.0.108/16
-phpldapadmin_hostname: ldap.novatech-llc.com
-phpldapadmin_port_args:
-    - "{{ phpldapadmin_ip_addr | ipaddr('address') }}:80:80"
-    - "{{ phpldapadmin_ip_addr | ipaddr('address') }}:443:443"
-
-phpmyadmin_ip_addr: 192.168.0.109/16
-phpmyadmin_hostname: phpmyadmin.novatech-llc.com
-
-buildbot_ip_addr: 192.168.0.110/16
-buildbot_hostname: buildbot.novatech-llc.com
-buildbot_port_args:
-    - "{{ buildbot_ip_addr | ipaddr('address') }}:80:80"
-    - "{{ buildbot_ip_addr | ipaddr('address') }}:443:443"
+daedalus_subnet: 192.168.0.0/24
+daedalus_net_connection: "System enp0s8"
diff --git a/ansible-playbook/site.yml b/ansible-playbook/site.yml
index 20ca10b3..82d8ce8c 100644
--- a/ansible-playbook/site.yml
+++ b/ansible-playbook/site.yml
@@ -51,6 +51,16 @@
     role: docker-common
     tags:
     - docker_common
+    services:
+      - svn
+      - openldap
+      - exim4
+      - gitlab
+      - wiki
+      - mantisbt
+      - build
+      - testssh
+      - buildbot
   - name: Configure Docker networking
     role: docker-networking
     tags:
diff --git a/roles/buildbot-master/defaults/main.yml b/roles/buildbot-master/defaults/main.yml
index 809062aa..5fee835d 100644
--- a/roles/buildbot-master/defaults/main.yml
+++ b/roles/buildbot-master/defaults/main.yml
@@ -2,16 +2,11 @@
 # defaults file for buildbot
 
 # The hostname passed as an envirnment variable into the container
-container_addr_map:
-  buildbot:
-    ip_addr: '127.0.0.1'
-    hostname: buildbot.example.com
-
-# the port mappings used when starting the container
-container_port_map:
-  buildbot:
-    port_args: 
-      - '{{ container_addr_map.buildbot.ip_addr }}:443:443'
+buildbot_ip_addr: '127.0.0.1'
+buildbot_hostname: buildbot.example.com
+buildbot_port_args: 
+  - '80'
+  - '443'
 
 # Network names
 buildbot_networks:
@@ -45,7 +40,7 @@ buildbot_volumes:
 buildbot_env:
   BUILDBOT_DATABASE: 'sqlite:///{{ buildbot_data_path }}/state.sqlite'
   BUILDBOT_WEB_PORT: '80'
-  BUILDBOT_WEB_URL: 'http://{{ container_addr_map.buildbot.hostname }}/'
+  BUILDBOT_WEB_URL: 'http://{{ buildbot_hostname }}/'
   BUILDBOT_SMTP_HOST: 'relay.dnsexit.com'
   BUILDBOT_SMTP_USER: 'Dean.cox'
   BUILDBOT_SMTP_PASS: 'no5156'
diff --git a/roles/buildbot-master/tasks/main.yml b/roles/buildbot-master/tasks/main.yml
index 88501072..94a1884f 100644
--- a/roles/buildbot-master/tasks/main.yml
+++ b/roles/buildbot-master/tasks/main.yml
@@ -31,4 +31,4 @@
     networks: '{{ buildbot_networks }}'
     volumes: '{{ buildbot_volumes }}'
     env: '{{ buildbot_env }}'
-    ports: '{{ container_port_map.buildbot.port_args }}'
+    ports: '{{ buildbot_port_args }}'
diff --git a/roles/docker-buildsystem/defaults/main.yml b/roles/docker-buildsystem/defaults/main.yml
index e91d9afb..b4341aa0 100644
--- a/roles/docker-buildsystem/defaults/main.yml
+++ b/roles/docker-buildsystem/defaults/main.yml
@@ -1,19 +1,19 @@
 # Variables that are used by the role, but defined at the playbook level.
 # docker_projects_dir
-# build_network.ip_addr
-# testssh_network.ip_addr
+# build_ip_addr
+# testssh_ip_addr
 # buildsystem_ssh_prod_user_password
 # buildsystem_ssh_root_user_password
 # buildsystem_database_root_password
 
 buildsystem:
     port_args: 
-      - '{{ build_network.ip_addr }}:80:80'
-      - '{{ build_network.ip_addr }}:443:443'
+      - '80'
+      - '443'
 
 test_ssh:
     port_args:
-    - '{{ testssh_network.ip_addr }}:2200:22'
+    - '22'
 
 # restore directories to temporarly store data being restored into docker containers
 backup_buildsystem_docker_dir: '{{ docker_backup_dir }}/buildsystem'
diff --git a/roles/docker-buildsystem/tasks/docker_containers/server_database.yml b/roles/docker-buildsystem/tasks/docker_containers/server_database.yml
index 31b3f33c..694f4144 100644
--- a/roles/docker-buildsystem/tasks/docker_containers/server_database.yml
+++ b/roles/docker-buildsystem/tasks/docker_containers/server_database.yml
@@ -25,7 +25,7 @@
     detach: True
     image: "{{ name_prefix }}teststation_database_server_image"
     name: "{{ name_prefix }}TestStation_database_server"
-    ports: "{{ build_network.ip_addr }}:3306:3306"
+    ports: "{{ build_ip_addr }}:3306:3306"
     state: "{{ container_start_state_nondata }}"
     restart_policy: no
     volumes_from:
diff --git a/roles/docker-buildsystem/tasks/docker_containers/server_http.yml b/roles/docker-buildsystem/tasks/docker_containers/server_http.yml
index 8b70f88f..6e3ad103 100644
--- a/roles/docker-buildsystem/tasks/docker_containers/server_http.yml
+++ b/roles/docker-buildsystem/tasks/docker_containers/server_http.yml
@@ -14,7 +14,7 @@
     image: "{{ name_prefix }}teststation_http_image"
     links: "{{ name_prefix }}TestStation_database_server:test_station_mysql_server"
     name: "{{ name_prefix }}TestStation_http_server"
-    ports: "{{ build_network.ip_addr }}:80:80"
+    ports: "{{ build_ip_addr }}:80:80"
     state: "{{ container_start_state_nondata }}"
     restart_policy: no
     volumes_from:
diff --git a/roles/docker-buildsystem/tasks/docker_containers/server_ssh.yml b/roles/docker-buildsystem/tasks/docker_containers/server_ssh.yml
index c7a659f0..b4512ea9 100644
--- a/roles/docker-buildsystem/tasks/docker_containers/server_ssh.yml
+++ b/roles/docker-buildsystem/tasks/docker_containers/server_ssh.yml
@@ -20,7 +20,7 @@
     image: "{{ name_prefix }}teststation_ssh_image"
     links: "{{ name_prefix }}TestStation_database_server:test_station_mysql_server"
     name: "{{ name_prefix }}TestStation_ssh_server"
-    ports: "{{ build_network.ip_addr }}:22:22"
+    ports: "{{ build_ip_addr }}:22:22"
     restart_policy: no
     state: "{{ container_start_state_nondata }}"
     volumes_from:
diff --git a/roles/docker-buildsystem/tasks/docker_containers/server_ssh_prod.yml b/roles/docker-buildsystem/tasks/docker_containers/server_ssh_prod.yml
index a2124cac..4258e7cc 100644
--- a/roles/docker-buildsystem/tasks/docker_containers/server_ssh_prod.yml
+++ b/roles/docker-buildsystem/tasks/docker_containers/server_ssh_prod.yml
@@ -31,6 +31,6 @@
     detach: True
     image: "{{ name_prefix }}teststation_ssh_server_manualtest_image"
     name: "{{ name_prefix }}TestStation_ssh_server_manualtest"
-    ports: "{{ testssh_network.ip_addr }}:22:22"
+    ports: "{{ testssh_ip_addr }}:22:22"
     state: "{{ container_start_state_nondata }}"
     restart_policy: no
diff --git a/roles/docker-buildsystem/tasks/docker_containers/server_tftp.yml b/roles/docker-buildsystem/tasks/docker_containers/server_tftp.yml
index 3d1ed431..917ec2db 100644
--- a/roles/docker-buildsystem/tasks/docker_containers/server_tftp.yml
+++ b/roles/docker-buildsystem/tasks/docker_containers/server_tftp.yml
@@ -13,7 +13,7 @@
     detach: True
     image: "{{ name_prefix }}teststation_tftp_image"
     name: "{{ name_prefix }}TestStation_tftp_server"
-    ports: "{{ build_network.ip_addr }}:69:69/udp"
+    ports: "{{ build_ip_addr }}:69:69/udp"
     state: "{{ container_start_state_nondata }}"
     restart_policy: no
     volumes_from:
diff --git a/roles/docker-buildsystem/tasks/main.yml b/roles/docker-buildsystem/tasks/main.yml
index 5c778707..2b626b35 100644
--- a/roles/docker-buildsystem/tasks/main.yml
+++ b/roles/docker-buildsystem/tasks/main.yml
@@ -2,8 +2,8 @@
 - assert:
     that:
     - docker_projects_dir is defined
-    - build_network.ip_addr is defined
-    - testssh_network.ip_addr is defined
+    - build_ip_addr is defined
+    - testssh_ip_addr is defined
     - buildsystem_ssh_prod_user_password is defined
     - buildsystem_ssh_root_user_password is defined
     - buildsystem_database_root_password is defined
diff --git a/roles/docker-common/tasks/main.yml b/roles/docker-common/tasks/main.yml
index d2de5a35..da2d022b 100644
--- a/roles/docker-common/tasks/main.yml
+++ b/roles/docker-common/tasks/main.yml
@@ -4,8 +4,6 @@
 - import_tasks: engine-redhat.yml
   when: ansible_os_family == "RedHat"
 
-- import_tasks: ip_addr.yml
-
 - import_tasks: packages.yml
 
 - import_tasks: configure.yml
diff --git a/roles/docker-exim4/defaults/main.yml b/roles/docker-exim4/defaults/main.yml
index 20108993..dd4c2eca 100644
--- a/roles/docker-exim4/defaults/main.yml
+++ b/roles/docker-exim4/defaults/main.yml
@@ -1,13 +1,12 @@
 ---
 
 # The hostname passed as an envirnment variable into the container
-exim4_network:
-  ip_addr: '127.0.0.1'
-  hostname: mail.example.com
-  port_args: 
-    - '{{ exim4_network.ip_addr }}:25:25'
-    - '{{ exim4_network.ip_addr }}:465:465'
-    - '{{ exim4_network.ip_addr }}:587:587'
+exim4_ip_addr: '127.0.0.1'
+exim4_hostname: mail.example.com
+exim4_port_args: 
+  - '25'
+  - '465'
+  - '587'
 
 # the repository containing Dockerfile to build the image
 exim4_image_repo: https://github.com/novatechweb/docker-exim4.git
diff --git a/roles/docker-exim4/tasks/main.yml b/roles/docker-exim4/tasks/main.yml
index 4760f5c2..4594a26a 100644
--- a/roles/docker-exim4/tasks/main.yml
+++ b/roles/docker-exim4/tasks/main.yml
@@ -106,10 +106,10 @@
 - name: start container
   docker_container:
     name: '{{ exim4_container_name }}'
-    hostname: '{{ exim4_network.hostname }}'
+    hostname: '{{ exim4_hostname }}'
     networks:
       - name: '{{ docker_network_frontend }}'
-    ports: '{{ exim4_network.port_args }}'
+    ports: '{{ exim4_port_args }}'
     image: '{{ exim4_image_name }}:{{ docker_image_tag }}'
     restart_policy: no
     state: started
diff --git a/roles/docker-gitlab/defaults/main.yml b/roles/docker-gitlab/defaults/main.yml
index e4268e33..d1931f58 100644
--- a/roles/docker-gitlab/defaults/main.yml
+++ b/roles/docker-gitlab/defaults/main.yml
@@ -1,13 +1,12 @@
 ---
 
 # The hostname passed as an envirnment variable into the container
-git_network:
-  ip_addr: '127.0.0.1'
-  hostname: git.example.com
-  port_args: 
-    - '{{ gitlab_network.ip_addr }}:80:80'
-    - '{{ gitlab_network.ip_addr }}:443:443'
-    - '{{ gitlab_network.ip_addr }}:22:22'
+git_ip_addr: '127.0.0.1'
+git_hostname: git.example.com
+git_port_args: 
+  - '80'
+  - '443'
+  - '22'
 
 # database usernames and passwords
 gitlab_db_user: novatech
diff --git a/roles/docker-gitlab/tasks/main.yml b/roles/docker-gitlab/tasks/main.yml
index bab47a2c..2adb1104 100644
--- a/roles/docker-gitlab/tasks/main.yml
+++ b/roles/docker-gitlab/tasks/main.yml
@@ -178,8 +178,8 @@
     name: '{{ gitlab_container_name }}'
     detach: true
     restart_policy: no
-    hostname: '{{ gitlab_network.hostname }}'
-    ports: '{{ gitlab_network.port_args }}'
+    hostname: '{{ gitlab_hostname }}'
+    ports: '{{ gitlab_port_args }}'
     networks:
       - name: '{{ docker_network_frontend }}'
     volumes:
@@ -204,7 +204,7 @@
 - name: Wait for gitlab to be fully running
   wait_for:
     delay: 10
-    host: '{{ gitlab_network.ip_addr }}'
+    host: "{{ gitlab_ip_addr }}"
     port: 22
     state: started
     timeout: 180
diff --git a/roles/docker-gitlab/templates/gitlab.rb.j2 b/roles/docker-gitlab/templates/gitlab.rb.j2
index 39984189..6ccaa864 100644
--- a/roles/docker-gitlab/templates/gitlab.rb.j2
+++ b/roles/docker-gitlab/templates/gitlab.rb.j2
@@ -200,7 +200,7 @@ gitlab_rails['ldap_enabled'] = true
 gitlab_rails['ldap_servers'] = YAML.load <<-'EOS'
   main: # 'main' is the GitLab 'provider ID' of this LDAP server
     label: 'NovaTech'
-    host: '{{ ldap_hostname }}'
+    host: '{{ openldap_hostname }}'
     port: 389
     uid: 'uid'
     bind_dn: 'cn=proxyagent,dc=novatech'
diff --git a/roles/docker-mantisbt/defaults/main.yml b/roles/docker-mantisbt/defaults/main.yml
index 5908d24e..245b0d74 100644
--- a/roles/docker-mantisbt/defaults/main.yml
+++ b/roles/docker-mantisbt/defaults/main.yml
@@ -1,12 +1,12 @@
 ---
 
 # The hostname passed as an envirnment variable into the container
-mantisbt_network:
-  ip_addr: '127.0.0.1'
-  hostname: mantis.example.com
-  port_args:
-    - '{{ mantisbt_network.ip_addr }}:80:80'
-    - '{{ mantisbt_network.ip_addr }}:443:443'
+network:
+mantisbt_ip_addr: '127.0.0.1'
+mantisbt_hostname: mantis.example.com
+mantisbt_port_args:
+  - '80'
+  - '443'
 
 # database usernames and passwords
 mantisbt_db_user: novatech
diff --git a/roles/docker-mantisbt/tasks/main.yml b/roles/docker-mantisbt/tasks/main.yml
index 31f92287..1eadc86e 100644
--- a/roles/docker-mantisbt/tasks/main.yml
+++ b/roles/docker-mantisbt/tasks/main.yml
@@ -219,15 +219,15 @@
     name: '{{ mantisbt_container_name }}'
     detach: true
     restart_policy: no
-    hostname: '{{ mantisbt_network.hostname }}'
-    ports: '{{ mantisbt_network.port_args }}'
+    hostname: '{{ mantisbt_hostname }}'
+    ports: '{{ mantisbt_port_args }}'
     networks:
       - name: '{{ docker_network_frontend }}'
     volumes: 
       - '{{ openssl_dv_name }}:/etc/ssl/private:z'
       - '{{ mantisbt_dv_name }}:/var/www/html:z'
     env: 
-      MANTISBT_HOSTNAME: '{{ mantisbt_network.hostname }}'
+      MANTISBT_HOSTNAME: '{{ mantisbt_hostname }}'
       MANTISBT_MAIL_USER: 'mantis'
       MANTISBT_MAIL_PASSWORD: '{{ mantisbt_email_password | quote }}'
       MANTISBT_LDAP_PASSWORD: '{{ openldap_proxyagent_password | quote }}'
diff --git a/roles/docker-mediawiki/defaults/main.yml b/roles/docker-mediawiki/defaults/main.yml
index 347f9f7c..437ef7c7 100644
--- a/roles/docker-mediawiki/defaults/main.yml
+++ b/roles/docker-mediawiki/defaults/main.yml
@@ -1,12 +1,11 @@
 ---
 
 # The hostname passed as an envirnment variable into the container
-wiki_network:
-  ip_addr: '127.0.0.1'
-  hostname: wiki.example.com
-  port_args:
-    - '{{ wiki_network.ip_addr }}:80:80'
-    - '{{ wiki_network.ip_addr }}:443:443'
+wiki_ip_addr: '127.0.0.1'
+wiki_hostname: wiki.example.com
+wiki_port_args:
+  - '80'
+  - '443'
 
 # database usernames and passwords
 wiki_db_user: novatech
diff --git a/roles/docker-mediawiki/tasks/main.yml b/roles/docker-mediawiki/tasks/main.yml
index c179d7cb..5ca42dbc 100644
--- a/roles/docker-mediawiki/tasks/main.yml
+++ b/roles/docker-mediawiki/tasks/main.yml
@@ -147,9 +147,9 @@
     delay: 3
     until: "container_status.rc == 0"
   - name: remove initialization container
-  docker_container:
+    docker_container:
       name: '{{ wiki_db_dv_name}}_init_db'
-    state: stopped
+      state: stopped
   - name: State file
     shell: 'date --rfc-3339=seconds > {{ docker_restore_config_base_dir }}/{{ wiki_db_dv_name }}/restore.date.txt'
 
@@ -172,15 +172,15 @@
     name: '{{ wiki_container_name }}'
     detach: true
     restart_policy: no
-    hostname: '{{ wiki_network.hostname }}'
-    ports: '{{ wiki_network.port_args }}'
+    hostname: '{{ wiki_hostname }}'
+    ports: '{{ wiki_port_args }}'
     networks:
       - name: '{{ docker_network_frontend }}'
     volumes: 
       - '{{ openssl_dv_name }}:/etc/ssl/private:z'
       - '{{ wiki_dv_name }}/var/www/html:z'
     env:
-      WIKI_HOSTNAME: '{{ container_addr_map.wiki.hostname }}'
+      WIKI_HOSTNAME: '{{ wiki_hostname }}'
       WIKI_MAIL_USER: 'wiki'
       WIKI_MAIL_PASSWORD: '{{ wiki_email_password | quote }}'
     image: '{{ wiki_image_name }}:{{ docker_image_tag }}'
diff --git a/roles/docker-common/tasks/ip_addr.yml b/roles/docker-networking/tasks/ip_addr.yml
similarity index 82%
rename from roles/docker-common/tasks/ip_addr.yml
rename to roles/docker-networking/tasks/ip_addr.yml
index 5a98cc37..4fe13727 100644
--- a/roles/docker-common/tasks/ip_addr.yml
+++ b/roles/docker-networking/tasks/ip_addr.yml
@@ -20,9 +20,12 @@
   register: network_pre
 
 - name: Add service IP addresses
-  command: nmcli connection modify '{{ network_connection }}' +ipv4.addresses '{{ item.value.ip_addr }}/16'
-  with_dict: "{{ container_addr_map }}"
-    
+  command: >
+    nmcli connection 
+    modify '{{ network_connection }}' 
+    +ipv4.addresses '{{ lookup('vars', item + "_ip_addr") }}/{{ subnet | ipaddr("prefix") }}'
+  loop: "{{ services }}"
+ 
 - sysctl:
     name: net.ipv4.ip_forward
     value: 1
diff --git a/roles/docker-networking/tasks/main.yml b/roles/docker-networking/tasks/main.yml
index 595e8bb7..1b1569c6 100644
--- a/roles/docker-networking/tasks/main.yml
+++ b/roles/docker-networking/tasks/main.yml
@@ -1,6 +1,14 @@
 ---
 # tasks file for docker-networking
 
+- assert:
+    that:
+    - net_connection is defined
+    - subnet is defined
+    - services is defined
+
+- import_tasks: ip_addr.yml
+
 - name: Create frontend network for services
   docker_network:
     name: '{{ docker_network_frontend }}'
diff --git a/roles/docker-openldap/defaults/main.yml b/roles/docker-openldap/defaults/main.yml
index 5dadcdcd..d1875f90 100644
--- a/roles/docker-openldap/defaults/main.yml
+++ b/roles/docker-openldap/defaults/main.yml
@@ -1,19 +1,17 @@
 ---
 
 # The hostname passed as an envirnment variable into the container
-openldap_network:
-  ip_addr: '127.0.0.1'
-  hostname: ldap.example.com
-  port_args:
-    - '{{ openldap_network.ip_addr }}:389:389'
-    - '{{ openldap_network.ip_addr }}:636:636'
-
-phpldapadmin_network:
-  ip_addr: '127.0.0.1'
-  hostname: ldap.example.com
-  port_args: 
-    - '{{ phpldapadmin_network.ip_addr }}:80:80'
-    - '{{ phpldapadmin_network.ip_addr }}:443:443'
+openldap_ip_addr: '127.0.0.1'
+openldap_hostname: ldap.example.com
+openldap_port_args:
+  - '389'
+  - '636'
+
+phpldapadmin_ip_addr: '127.0.0.1'
+phpldapadmin_hostname: ldap.example.com
+phpldapadmin_port_args: 
+  - '80'
+  - '443'
 
 openldap_backup_files:
   - ldif_file: LDAP_database_0000.ldif
diff --git a/roles/docker-openldap/tasks/main.yml b/roles/docker-openldap/tasks/main.yml
index 070a494a..54d543f0 100644
--- a/roles/docker-openldap/tasks/main.yml
+++ b/roles/docker-openldap/tasks/main.yml
@@ -112,14 +112,14 @@
     name: '{{ openldap_container_name }}'
     detach: true
     restart_policy: no
-    hostname: '{{ openldap_network.hostname }}'
-    ports: '{{ ldap_network.port_args }}'
+    hostname: '{{ openldap_hostname }}'
+    ports: '{{ openldap_port_args }}'
     volumes:
       - '{{ openssl_dv_name }}:/etc/ssl/private:z'
       - '{{ openldap_cv_name }}:/etc/ldap:z'
       - '{{ openldap_dv_name }}:/var/lib/ldap:z'
     env:
-      LDAP_HOSTNAME: '{{ openldap_network.hostname }}'
+      LDAP_HOSTNAME: '{{ openldap_hostname }}'
     image: '{{ openldap_image_name }}:{{ docker_image_tag }}'
     networks:
       - name: '{{ docker_network_frontend }}'
@@ -157,14 +157,14 @@
     name: '{{ phpldapadmin_container_name }}'
     detach: true
     restart_policy: no
-    hostname: '{{ phpldapadmin_network.hostname }}'
-    ports: '{{ phpldapadmin_network.port_args }}'
+    hostname: '{{ phpldapadmin_hostname }}'
+    ports: '{{ phpldapadmin_port_args }}'
     volumes:
       - '{{ openssl_dv_name }}:/etc/ssl/private:z'
     env:
       PHPLDAPADMIN_HTTPS: 'false'
       PHPLDAPADMIN_LDAP_HOSTS: 'ldap'
-      LDAP_HOSTNAME: '{{ container_addr_map.phpldapadmin.hostname }}'
+      LDAP_HOSTNAME: '{{ openldap_hostname }}'
     networks:
       - name: '{{ docker_network_frontend }}'
     image: '{{ phpldapadmin_image_name }}:{{ docker_image_tag }}'
diff --git a/roles/docker-svn/defaults/main.yml b/roles/docker-svn/defaults/main.yml
index e3e85d16..93ba7053 100644
--- a/roles/docker-svn/defaults/main.yml
+++ b/roles/docker-svn/defaults/main.yml
@@ -1,12 +1,11 @@
 ---
 
 # The hostname passed as an envirnment variable into the container
-svn_network:
-  ip_addr: '127.0.0.1'
-  hostname: svn.example.com
-  port_args: 
-    - '{{ svn_network.ip_addr }}:80:80'
-    - '{{ svn_network.ip_addr }}:443:443'
+svn_ip_addr: '127.0.0.1'
+svn_hostname: svn.example.com
+svn_port_args: 
+  - '80'
+  - '443'
 
 # The repositories that will be created/imported durring a data-volume restore
 svn_repos: 'example'
diff --git a/roles/docker-svn/tasks/main.yml b/roles/docker-svn/tasks/main.yml
index 89c74d91..9b7e0bbd 100644
--- a/roles/docker-svn/tasks/main.yml
+++ b/roles/docker-svn/tasks/main.yml
@@ -81,15 +81,15 @@
     name: '{{ svn_container_name }}'
     detach: true
     restart_policy: no
-    hostname: '{{ svn_network.hostname }}'
-    ports: '{{ svn_network.port_args }}'
+    hostname: '{{ svn_hostname }}'
+    ports: '{{ svn_port_args }}'
     networks:
       - name: '{{ docker_network_frontend }}'
     volumes:
       - '{{ openssl_dv_name }}:/etc/ssl/private:z'
       - '{{ svn_dv_name }}:/var/lib/svn:z'
     env:
-      SVN_HOSTNAME: '{{ svn_network.hostname }}'
+      SVN_HOSTNAME: '{{ svn_hostname }}'
       SVN_LDAP_PASSWORD: '{{ openldap_proxyagent_password }}'
     image: '{{ svn_image_name }}:{{ docker_image_tag }}'
